### DISCUSSION TITLE
Is there no way to re-run ./boostrap_vcpkg.sh with out the internet?

### DISCUSSION THREAD

**Discussion by Cazadorro** [microsoft/vcpkg#37695] [Q&A]
I'm trying to deploy a vcpkg --only-download on a offline system.  I'm getting errors about it not finding a git version above 2.7.4, which is patently false, git --version displays 2.27, it's set in .bashrc as well.  My best guess is that because I ran boostrap_vcpkg.sh on a system with a different version of git (2.9) that it just just not finding that and complaining?  A hypothetical fix is to run bootstrap_vcpkg.sh again, however, I can't do that because it instantly attempts to download things from the internet.  The thing is every dependency it should be downloading *should already be downloaded*, so there's no reason for it to try to download again.   This post https://github.com/microsoft/vcpkg/issues/31641 seems to imply that I shouldn't be seeing any issues here *period*, since I already have things in the right place by definition, my vcpkg directory is a straight copy from a system where it *worked*.   Is there a way to get around these issues?

**Comment by baebrum** [comment]
Did you find a solution to the issue? I am running into a similar issue where vcpkg install without or without manifest mode insists on reaching out to my private registry instead of checking locally. I haven't found a workaround or solution yet.

**Reply by Cazadorro** [reply]
Private registry is a different issue, I just had this issue and got around this by just having overlay ports of everything I want, and getting those offline independently (and setting the git URL in my port files to point to local file directories), but AFAIK you *should* be able to get around this by changing your registry address in your vcpkg-configuration.json to a local file directory.     I don't see how to make this "configurable" thus not require code changes every time you're in an offline environment, since registries aren't exposed as CMakePreset variables nor are environment variables usable in vcpkg-configuration.json, so that's a little annoying, but in theory you only have to edit this in a single place.   I don't think there's been a proper support of registries in download-only let alone a discussion for VCPKG.
